src/api/db/models/Project.ts

```diff
@@ -0,0 +1,746 @@
+          const deployment = camConfig!.deployments.find((dep) => idMatch(dep._id!, deploymentId));
+          if (deployment!.name === 'default') {
+            bail(new ForbiddenError(`View ${deployment!.name} is not editable`));
+          }
+
+          // apply updates, sort deployments, and save project
+          Object.assign(deployment!, diffs);
+          camConfig.deployments = sortDeps(camConfig.deployments);
+          await project.save();
+          return { project, camConfig };
+        },
+        { retries: 2 },
+      );
+      // TODO: we need to reverse the above operation if reMapImagesToDeps fails!
+      if (Object.keys(input.diffs).includes('startDate')) {
+        await ProjectModel.reMapImagesToDeps({ projId: project._id, camConfig });
+      }
+      return camConfig;
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async deleteDeploymentTask(
+    input: gql.DeleteDeploymentInput,
+    context: Context,
+  ): Promise<HydratedDocument<TaskSchema>> {
+    try {
+      return await TaskModel.create(
+        {
+          type: 'DeleteDeployment',
+          projectId: context.user['curr_project'],
+          user: context.user.sub,
+          config: input,
+        },
+        context,
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  // NOTE: this function is called by the task handler
+  static async deleteDeployment(
+    { cameraId, deploymentId }: gql.DeleteDeploymentInput,
+    context: Context,
+  ): Promise<HydratedCameraConfig> {
+    try {
+      const { project, camConfig } = await retry(
+        async () => {
+          // find camera config
+          const [project] = await ProjectModel.getProjects(
+            { _ids: [context.user['curr_project']!] },
+            context,
+          );
+          const camConfig = project.cameraConfigs.find((cc) => idMatch(cc._id, cameraId))!;
+
+          // filter out deployment, sort remaining ones, and save project
+          camConfig.deployments = sortDeps(
+            camConfig!.deployments.filter(
+              (dep) => !idMatch(dep._id!, deploymentId),
+            ) as typeof camConfig.deployments,
+          );
+
+          await project.save();
+          return { project, camConfig };
+        },
+        { retries: 2 },
+      );
+      // TODO: we need to reverse the above operation if reMapImagesToDeps fails!
+      await ProjectModel.reMapImagesToDeps({ projId: project._id, camConfig });
+      return camConfig;
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async createLabel(
+    input: gql.CreateProjectLabelInput,
+    context: Context,
+  ): Promise<HydratedProject['labels'][0]> {
```

Comment:
In this case, because we don't appear to nee to use any of Mongoose methods on this subdocument, I think it's ok to return the "Nested Path" type rather than the Hydrated Subdocument type? Perhaps that's true in other places as well.

Reply:
No change requested just an observation.

Reply:
I think I was just being a bit sloppy at that moment, properly typing the output as done in [b802430](https://github.com/tnc-ca-geo/animl-api/pull/220/commits/b8024302211ff73c8b8f073c8d36652147634ee6) is probably the more correct way for us to go.